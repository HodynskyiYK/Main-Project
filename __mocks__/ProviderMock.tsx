import React from 'react'
import {MemoryRouter} from 'react-router-dom'
import { MovieModalContextProvider } from '../src/context/movie-modal-context'
import { Provider } from 'react-redux'
import { store } from '../src/store'
import { MovieDetailsContextProvider } from '../src/context/movie-details-context'

// @ts-ignore
export const ProvidersMock = ({ children }) => {
    return (
        <Provider store={store}>
            <MovieModalContextProvider>
                <MovieDetailsContextProvider>
                    <MemoryRouter>{children}</MemoryRouter>
                </MovieDetailsContextProvider>
            </MovieModalContextProvider>
        </Provider>
    )
}