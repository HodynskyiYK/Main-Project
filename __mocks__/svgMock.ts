// Mock Svg files
// https://github.com/gregberge/svgr/issues/83#issuecomment-437342861
const SvgMock = "icon-mock"
export const ReactComponent = SvgMock
export default SvgMock