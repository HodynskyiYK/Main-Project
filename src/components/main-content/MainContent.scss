.main-content {
  background: $dark_grey;

  .top-content {
    position: relative;
    border-bottom: 1px solid #080707;
    z-index: 1;

    &:after {
      content: '';
      position: absolute;
      bottom: 0;
      left: 0;
      width: 100%;
      height: 0.25rem;
      background: $grey;
    }
  }
}