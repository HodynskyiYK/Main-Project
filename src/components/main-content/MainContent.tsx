import React, {FC} from 'react'
import {FILTERS_VALUES, MoviesFilters} from '../movies-filters'
import {MovieSorting} from '../movie-sorting'
import {MOVIES, MoviesList} from '../movies-list'

export const MainContent: FC = () => {

    return (
        <main>
            <section>
                <div className="main-content container">
                    <div className="top-content">
                        <div className="row align-items-center justify-content-between">
                            <div className="col">
                                <MoviesFilters filters={FILTERS_VALUES}/>
                            </div>
                            <div className="col">
                                <MovieSorting/>
                            </div>
                        </div>
                    </div>
                    <MoviesList moviesList={MOVIES}/>
                </div>
            </section>
        </main>
    )
}
