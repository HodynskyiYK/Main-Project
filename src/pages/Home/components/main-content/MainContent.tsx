import React, { FC } from 'react'
import classnames from 'classnames'
import {FILTERS_VALUES, MoviesFilters} from '../movies-filters'
import {MovieSorting} from '../movie-sorting'
import { MoviesList } from '../movies-list'
import styles from './MainContent.module.scss'

export const MainContent: FC = () => {

    return (
        <main>
            <section>
                <div className={classnames('container', styles.mainContent)}>
                    <div className={styles.topContent}>
                        <div className={classnames('row', 'align-items-center', 'justify-content-between')}>
                            <div className={classnames('col')}>
                                <MoviesFilters filters={FILTERS_VALUES}/>
                            </div>
                            <div className={classnames('col')}>
                                <MovieSorting/>
                            </div>
                        </div>
                    </div>
                    <MoviesList/>
                </div>
            </section>
        </main>
    )
}
