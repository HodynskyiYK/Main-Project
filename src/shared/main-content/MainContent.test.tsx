import React from 'react'
import { render } from '@testing-library/react'
import { MainContent } from './MainContent'
import { ProvidersMock } from '../../../__mocks__/ProviderMock'

describe('MainContent tests', () => {

    test('To match snapshot', () => {
        const {asFragment} = render(<ProvidersMock><MainContent/></ProvidersMock>)
        expect(asFragment()).toMatchSnapshot()
    })
})