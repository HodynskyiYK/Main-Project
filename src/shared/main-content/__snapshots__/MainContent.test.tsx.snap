// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContent tests To match snapshot 1`] = `
<DocumentFragment>
  <main>
    <section>
      <div
        class="container mainContent"
      >
        <div
          class="topContent"
        >
          <div
            class="row align-items-center justify-content-between"
          >
            <div
              class="col"
            >
              <ul
                class="filtersList"
              >
                <li
                  class="filtersItem"
                >
                  <span
                    class="filterValue activeFilter"
                  >
                    All
                  </span>
                </li>
                <li
                  class="filtersItem"
                >
                  <span
                    class="filterValue"
                  >
                    Documentary
                  </span>
                </li>
                <li
                  class="filtersItem"
                >
                  <span
                    class="filterValue"
                  >
                    Comedy
                  </span>
                </li>
                <li
                  class="filtersItem"
                >
                  <span
                    class="filterValue"
                  >
                    Horror
                  </span>
                </li>
                <li
                  class="filtersItem"
                >
                  <span
                    class="filterValue"
                  >
                    Crime
                  </span>
                </li>
                <li
                  class="filtersItem"
                >
                  <span
                    class="filterValue"
                  >
                    Action
                  </span>
                </li>
              </ul>
            </div>
            <div
              class="col"
            >
              <div
                class="movieSorting"
              >
                <p
                  class="label"
                >
                  Sort by
                </p>
                <p
                  class="selectedValue"
                >
                  Release date
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="moviesCounter"
        >
          <p>
            <b>
              0
            </b>
             movies found
          </p>
        </div>
        <div
          class="loaderContent"
        >
          <span
            class="loader"
          />
        </div>
      </div>
    </section>
  </main>
</DocumentFragment>
`;
